types/src/json/tx.rs

```diff
@@ -0,0 +1,1455 @@
+///
+/// Used in:
+/// - [`/get_transactions` -> `txs.as_json`](https://www.getmonero.org/resources/developer-guides/daemon-rpc.html#get_transactions)
+/// - [`/get_transaction_pool` -> `tx_json`](https://www.getmonero.org/resources/developer-guides/daemon-rpc.html#get_transaction_pool)
+#[derive(Clone, Debug, PartialEq, Eq, PartialOrd, Ord, Hash)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+#[serde(untagged)]
+pub enum Transaction {
+    V1 {
+        /// This field is [flattened](https://serde.rs/field-attrs.html#flatten).
+        #[serde(flatten)]
+        prefix: TransactionPrefix,
+        signatures: Vec<HexBytes<64>>,
+    },
+    V2 {
+        /// This field is [flattened](https://serde.rs/field-attrs.html#flatten).
+        #[serde(flatten)]
+        prefix: TransactionPrefix,
+        rct_signatures: RctSignatures,
+        /// This field is [`Some`] if [`Self::V2::rct_signatures`]
+        /// is [`RctSignatures::NonCoinbase`], else [`None`].
+        rctsig_prunable: Option<RctSigPrunable>,
+    },
+}
+
+/// [`Transaction::V1::prefix`] & [`Transaction::V2::prefix`].
+#[derive(Clone, Debug, PartialEq, Eq, PartialOrd, Ord, Hash)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+pub struct TransactionPrefix {
+    pub version: u8,
+    pub unlock_time: u64,
+    pub vin: Vec<Input>,
+    pub vout: Vec<Output>,
+    pub extra: Vec<u8>,
+}
+
+impl From<transaction::Transaction> for Transaction {
+    fn from(tx: transaction::Transaction) -> Self {
+        fn map_prefix(prefix: transaction::TransactionPrefix, version: u8) -> TransactionPrefix {
+            let mut height = 0;
+
+            let vin = prefix
+                .inputs
+                .into_iter()
+                .filter_map(|input| match input {
+                    transaction::Input::ToKey {
+                        amount,
+                        key_offsets,
+                        key_image,
+                    } => {
+                        let key = Key {
+                            amount: amount.unwrap_or(0),
+                            key_offsets,
+                            k_image: HexBytes::<32>(key_image.compress().0),
+                        };
+
+                        Some(Input { key })
+                    }
+                    transaction::Input::Gen(h) => {
+                        height = usize_to_u64(h);
+                        None
+                    }
+                })
+                .collect();
+
+            let vout = prefix
+                .outputs
+                .into_iter()
+                .map(|o| {
+                    let amount = o.amount.unwrap_or(0);
+
+                    let target = match o.view_tag {
+                        Some(view_tag) => {
+                            let tagged_key = TaggedKey {
+                                key: HexBytes::<32>(o.key.0),
+                                view_tag: HexBytes::<1>([view_tag]),
```

Comment:
Nit: you shouldn't need to specify the const generic on every use Rust should be able to infer it